lib/game_team.rb

```diff
@@ -37,6 +37,30 @@ def self.winningest_coach
   def self.worst_coach
   end
 
+  def self.best_offense
+    #grouped by team_id with keys being the team_id and values is an array of games
+    grouped_team = @@all.group_by{|game| game.team_id}
+    #loop through the values (games) and set them equal to the average of goals
+    team_averaged_goals = grouped_team.map do |ids, games|
+      goals_per_game = games.map {|game| game.goals}
+      games = (goals_per_game.sum / goals_per_game.length)
+    end
+    total_goals_per_team= Hash[grouped_team.keys.zip(team_averaged_goals)]
+    total_goals_per_team.key(total_goals_per_team.values.max)
+  end
+
```

Comment:
I think this is how I *should* be writing worst_coach and winningest_coach, so once we get this merged and pulled down I'll try it out. Nice!